lib/metrics/statsd.js

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var StatsD = require('node-statsd')
+
+var STATSD_PREFIX = 'fxa.auth.'
+
+function StatsDCollector(log) {
+  if (! log) {
+    throw new Error('Log is required')
+  }
+
+  var config = require('../../config')
+  var statsdConfig = config.get('statsd')
+
+  this.host = statsdConfig.host
+  this.port = statsdConfig.port
+  this.sampleRate = statsdConfig.sample_rate
+  this.connected = false
+  this.client = null
+  this.log = log
+}
+
+StatsDCollector.prototype = {
+  /**
+   * Initializes a StatsD socket client
+   */
+  init: function () {
+    var self = this
+    var client = this.client = new StatsD(this.host, this.port)
+
+    if (client.socket) {
+      this.connected = true
+      client.socket.on('error', function (error) {
+        self.log.error({ op: 'statsd', err: error })
```

Comment:
Should we flip `this.connected` back to `false` on error?  Or does it recover automatically and stay connected?